tmc-langs-framework/src/main/java/fi/helsinki/cs/tmc/langs/ExerciseBuilder.java

```diff
@@ -0,0 +1,122 @@
+package fi.helsinki.cs.tmc.langs;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import org.apache.commons.io.FileUtils;
+
+public class ExerciseBuilder {
+
+    private String beginSolution = "// BEGIN SOLUTION";
+    private String endSolution = "// END SOLUTION";
+    private String stubMarker = "// STUB:";
+    private String sourceFolderName = "src";
+
+    /**
+     * Constuctor
+     *
+     * <p>
+     * TODO: Needs to be expanded with LanguagePlugin, which determines the
+     * comment style used with the language
+     *
+     */
+    public ExerciseBuilder() {
+
+    }
+
+    /**
+     * Prepares a stub exercise from the original.
+     *
+     * <p>
+     * The stub is a copy of the original where the model solution and special
+     * comments have been stripped and stubs like ('return 0') have been added.
+     *
+     * @param path A path to a directory where the original exericse has been
+     * copied. This method should modify the contents of this directory.
+     */
```

Comment:
Instead of duplicating the comment, maybe just say "Implements LanguagePlugin.prepareStub".

If this were Scala or Java 8, this class would probably make a nice mixin for LanguagePlugin.